filprofiler/_tracer.py

```diff
@@ -23,10 +25,9 @@
 preload.fil_initialize_from_python()
 
 
-def start_tracing(output_path: str):
+def start_tracing(output_path: Union[str, Path]):
```

Comment:
In my view it's fine to use paths, but I wonder if it would make sense to return a summary data structure in the absence of a path, akin to memory-profiler [memory_usage.](https://github.com/pythonprofilers/memory_profiler/blob/master/memory_profiler.py#L271)

There are some advantages to how memory-profiler separates generating the profiling data from creating a report based on that data; this allows report generation to be deferred or customized.

Reply:
The raw data is actually written out too, in a format you can feed into `flamegraph.pl` or `inferno`.

What would you think would be useful in such a report?

Reply:
On closer inspection the raw data as it is written currently is probably fine.  Perhaps it would be enough to allow passing a file-like object (e.g a BytesIO buffer) as an alternative to a path, in which case just the raw data would be written to the file?

Reply:
2 goals here:
 - a disk independent mechanism to collect results
 - the ability to collect results without generating derived report artifacts
 
 These things make it easy for a user to control profiling in their script and then plot when it suits them.

Reply:
1. (Some) of the generating code is in Rust, so would have to write some bridge code if the goal is avoiding disk altogether.
2. The data format isn't stable yet, there's some terrible hacks I will eventually get rid of.
3. In general I'd like reports to be as good as possible by default, so would love ideas for improving it.
4. In the short term, you can always emulate this by just opening the files and copying them into wherever you want.

So maybe something to do later, but for now I'm just going to stick to this API.